### PR TITLE
Factor out portions of PreparedAddDoc to PreparedAddNode.

### PR DESCRIPTION
**Pull request by toregge**
@geirst : please review

### CODE REVIEW

File: searchlib/src/vespa/searchlib/tensor/hnsw_index.cpp [hunk 1]
```diff
@@ -415,16 +436,26 @@ HnswIndex::filter_valid_nodeids(uint32_t level, const PreparedAddDoc::Links &nei
...
-    auto nodeid = nodeids[0];
-    auto node_ref = _graph.make_node(nodeid, docid, 0, op.max_level + 1);
-    for (int level = 0; level <= op.max_level; ++level) {
-        auto neighbors = filter_valid_nodeids(level, op.connections[level], nodeid);
+    assert(op.nodes.size() == 1);
+    auto nodeids = _id_mapping.allocate_ids(docid, op.nodes.size());
+    assert(nodeids.size() == op.nodes.size());
+    uint32_t subspace = 0;
+    for (auto nodeid : nodeids) {
+        internal_complete_add_node(nodeid, docid, subspace, op.nodes[subspace]);
```
Comment (geirst) [inline searchlib/src/vespa/searchlib/tensor/hnsw_index.cpp:444]: I guess subspace should be increased as well?

**Review by geirst — COMMENTED** [review]
👍

File: searchlib/src/vespa/searchlib/tensor/hnsw_index.cpp (hunk 1 above)
Comment (toregge) [reply searchlib/src/vespa/searchlib/tensor/hnsw_index.cpp:444]: Yes.

**Review by geirst — APPROVED** [review]
👍

**Pull request merged**